src/objective-c/GRPCClient/private/GRPCHost.m

```diff
@@ -127,8 +181,10 @@ - (GRPCChannel *)channel {
         args[@GRPC_SSL_TARGET_NAME_OVERRIDE_ARG] = _hostNameOverride;
       }
 
+      grpc_channel_credentials *creds = [self channelCredentials:_pemRootCerts privateKey:_pemPrivateKey certChain: _pemCertChain];
```

Comment:
`grpc_ssl_credentials_create` copies the content of the root certs, certificate chain, and private key. Because the majority of channels will use the default root certs, we want to have only a copy of the corresponding `grpc_channel_credentials` in memory (instead of one per channel).

For the non-default case it's not bad, because we have just one channel per host. But `-[GRPCHost channel]` is called lazily, only when the first RPC to that host is to be sent. It's preferable if we create the `grpc_channel_credentials` object eagerly instead: If there's something wrong with the certificates or the key, debugging is much easier if the failure shows up when the user calls `[GRPCCall set... forHost:...]`, rather than later on upon starting an RPC.